pkg/agent/manager/cache/cache.go

```diff
@@ -332,8 +327,19 @@ func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.Regi
 		// Determine if something related to this record changed outside of the
 		// selectors and if so, make sure subscribers for all entry selectors
 		// are notified.
-		if federatedBundlesChanged {
-			notifySet.Merge(newEntry.Selectors...)
+		if federatedBundlesChanged || c.selectorsDifferent(existingEntry, newEntry) {
```

Comment:
I think the introduction of, and this call to, `selectorsDifferent` is redundant. One or both of `selAdd` and `selRem` should have len > 0 if there is a difference in selectors between the existing and new versions of the entry.

Also, the comment above this block needs to be updated to remove the "outside of the selectors" part.

Reply:
Thanks good catch. It flows much better now.